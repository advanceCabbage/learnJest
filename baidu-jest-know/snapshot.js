export const gennerateConfig = () => {
  return {
    server: "http://localhost",
    port: 8080,
    time: new Date(),
  };
};

export const gennerateAnthorConfig = () => {
  return {
    server: "http://localhost",
    port: 8080,
    time: 2021,
  };
};
