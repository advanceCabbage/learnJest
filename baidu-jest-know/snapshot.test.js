import { gennerateConfig, gennerateAnthorConfig } from "./snapshot";

//! toMatchSnapshot 适用于测试配置文件是否被修改场景
test("测试 gennerateConfig ", () => {
  expect(gennerateConfig()).toMatchSnapshot({
    time: expect.any(Date), //todo 表明这里的time无需相等，只需保证为Date类型
  });
});

test("测试 gennerateAnthorConfig ", () => {
  expect(gennerateAnthorConfig()).toMatchSnapshot();
});

test("测试 gennerateConfig ", () => {
  expect(gennerateConfig()).toMatchInlineSnapshot(
    //todo 行内快照 将快照生成到当前文件中
    {
      time: expect.any(Date),
    },
    `
    Object {
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
