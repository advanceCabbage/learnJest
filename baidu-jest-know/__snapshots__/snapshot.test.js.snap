// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 gennerateAnthorConfig  1`] = `
Object {
  "port": 8080,
  "server": "http://localhost",
  "time": 2021,
}
`;

exports[`测试 gennerateConfig  1`] = `
Object {
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
